src/main/java/com/artipie/api/KeyStore.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.api;
+
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.blocking.BlockingStorage;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PfxOptions;
+import java.util.List;
+
+/**
+ * Key store.
+ * @since 0.26
+ */
+public abstract class KeyStore {
+    /**
+     * YAML node name for `enabled` yaml section.
+     */
+    private static final String NODE_ENABLED = "enabled";
+
+    /**
+     * YAML node name for `path` yaml section.
+     */
+    private static final String PATH = "path";
+
+    /**
+     * YAML node name for `password` yaml section.
+     */
+    private static final String PASSWORD = "password";
+
+    /**
+     * YAML node name for `alias` yaml section.
+     */
+    private static final String ALIAS = "alias";
+
+    /**
+     * YAML node name for `aliasPassword` yaml section.
+     */
+    private static final String ALIAS_PASSWORD = "alias-password";
+
+    /**
+     * YAML-configuration of key store.
+     */
+    private final YamlMapping yml;
+
+    /**
+     * Ctor.
+     * @param yaml YAML.
+     */
+    public KeyStore(final YamlMapping yaml) {
+        this.yml = yaml;
+    }
+
+    /**
+     * Checks if SSL is enabled.
+     * @return True is SSL enabled.
+     */
+    public boolean enabled() {
+        return Boolean.parseBoolean(this.yml.string(KeyStore.NODE_ENABLED));
+    }
+
+    /**
+     * Create KeyStore instance.
+     * @param yaml Settings of key store
+     * @return KeyStore
+     */
+    @SuppressWarnings("PMD.ProhibitPublicStaticMethods")
+    public static KeyStore create(final YamlMapping yaml) {
+        final List<KeyStore> keystores = List.of(
+            new JksKeyStore(yaml), new PemKeyStore(yaml), new PfxKeyStore(yaml)
+        );
+        for (final KeyStore keystore : keystores) {
+            if (keystore.isConfigured()) {
+                return keystore;
+            }
+        }
+        throw new IllegalStateException("Not found configuration in 'ssl'-section of yaml");
+    }
+
+    /**
+     * Checks if configuration for this type of KeyStore is present.
+     * @return True if it is configured.
+     */
+    public abstract boolean isConfigured();
+
+    /**
+     * Create http server.
+     * @param vertx Vertx.
+     * @param storage Artipie settings storage.
+     * @return HttpServer
+     */
+    public abstract HttpServer createHttpServer(Vertx vertx, Storage storage);
```

Comment:
It looks strange that `KeyStore` creates `HttpServer`.
Suggest something like this:
```
public void updateOptions(
            final HttpServerOptions options,
            final Storage storage) {
            options.setPemKeyCertOptions(this.pemOptions(storage));
        }
```

Reply:
Vertx allows to pass options only creation time of HttpServer

Reply:
> Vertx allows to pass options only creation time of HttpServer

Ok, how does it influence on my comment?

Reply:
I got your point, added enchancement.